sqlite3_go18_test.go

```diff
@@ -136,6 +136,89 @@ func TestShortTimeout(t *testing.T) {
 	}
 }
 
+func TestBeginTxCancel(t *testing.T) {
+	srcTempFilename := TempFilename(t)
+	defer os.Remove(srcTempFilename)
+
+	db, err := sql.Open("sqlite3", srcTempFilename)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	db.SetMaxOpenConns(10)
+	db.SetMaxIdleConns(5)
+
+	defer db.Close()
+	initDatabase(t, db, 100)
+
+	wg := sync.WaitGroup{}
+	// create several go-routines to expose racy issue
+	for i := 0; i < 10; i++ {
```

Comment:
What is the need for this outer for loop? Why is the inner one insufficient?

Reply:
inner loop is inside go-routine

Reply:
Hmm, the only thing I can think of is that the extra goroutines cause the scheduler to behave slightly differently. For example, if it causes the `cancel` goroutine to consistently complete before the `execSync` goroutine, that would explain the discrepancy. (The old code would have consistently hit the "no need to interrupt" case.)

That does give me an idea for something more deterministic here. If we were to call `BeginTx` directly on `*SQliteConn` (rather than going through the database/sql package) with an already canceled context, that would have almost always caused the failure in the old code, but now will always work as expected.

Reply:
I did as you suggested and it really much more deterministic (still not 100% as we want to interrupt statement with sqlite3_interrupt.

From tests I see that Conn.Raw() method was introduced only in go 1.13. Will move that test separately.